addons/im_livechat/report/im_livechat_report_agent_history_views.xml

```diff
@@ -0,0 +1,105 @@
+                    <field name="rating" string="Rating (%)" type="measure" widget="im_livechat.rating_percentage"/>
+                    <field name="response_time_hour" string="Response Time" type="measure" widget="float_time" />
+                    <field name="session_duration_hour" type="measure" widget="float_time"/>
+                </graph>
+            </field>
+        </record>
+
+        <record id="im_livechat_report_agent_history_view_pivot" model="ir.ui.view">
+            <field name="name">im_livechat.report.agent.history.pivot</field>
+            <field name="model">im_livechat.channel.member.history</field>
+            <field name="arch" type="xml">
+                <pivot string="Livechat Support Statistics" sample="1" display_quantity="1">
+                    <field name="has_call" invisible="1"/>
+                    <field name="session_week_day" invisible="1"/>
+                    <field name="partner_id" type="row"/>
+                    <field name="call_duration_hour" type="measure" widget="float_time" options="{'displaySeconds': True}"/>
+                    <field name="call_percentage" type="measure" widget="percentage"/>
+                    <field name="response_time_hour" string="Response Time (hh:mm:ss)" type="measure" widget="float_time" options="{'displaySeconds': True}" />
+                    <field name="session_duration_hour" type="measure" widget="float_time" options="{'displaySeconds': True}"/>
+                    <field name="rating" string="Rating (%)" type="measure" widget="im_livechat.rating_percentage"/>
+                    <field name="call_count" type="measure" widget="integer"/>
+                </pivot>
+            </field>
+        </record>
+        <record id="im_livechat_report_agent_history_action" model="ir.actions.act_window">
+            <field name="name">Agents</field>
+            <field name="res_model">im_livechat.channel.member.history</field>
+            <field name="view_mode">pivot,graph</field>
+            <field name="domain">[('livechat_member_type', '=', 'agent')]</field>
+            <field name="context">
+                {
+                    "search_default_group_by_agent": 1,
+                    "search_default_filter_date_last_month": 1,
+                    "graph_measure": "__count__",
+                    "pivot_measures": ["__count", "response_time_hour", "session_duration_hour", "rating", "call_count"],
+                    "im_livechat.hide_partner_company": True
+                }
+            </field>
+            <field name="search_view_id" ref="im_livechat_report_agent_history_view_search"/>
+            <field name="help" type="html">
+                <p class="o_view_nocontent_empty_folder">No data yet!</p>
+                <p>Track and improve live chat performance with insights on session activity, response times, customer ratings, and call interactions.</p>
+            </field>
+        </record>
```

Comment:
When clicking on a row, it opens the list view of the history. I think we might want to open the list view of related channels instead?

Then if clicking on the row of the list view, we open the history form, which is currently not done.

Reply:
Only for pivot, so does the report channel view. I fixed both at once.